client/pages/CalendarScreen/EventManager/DescriptionInput.tsx

```diff
@@ -0,0 +1,91 @@
+import { eventNames } from 'process';
+import React from 'react';
+import { Button, Modal, StyleSheet, Text, Pressable, View, TextInput } from 'react-native';
+import DateTimePicker from '@react-native-community/datetimepicker';
+
+const ModalPopup = ({ visible, save, cancel, children }) => {
+    const [showModal, setShowModal] = React.useState(visible);
+
+    return (
+        <Modal transparent visible={true}>
+            <View style={styles.modalBackground}>
+                <View style={[styles.modalContainer]}>
+                    <View style={styles.header}>
+                        <Button title="Cancel" onPress={cancel} />
+                        <Button title="Save" onPress={save} />
+                    </View>
+                    {children}
+                </View>
+            </View>
+        </Modal>
+    );
+};
+
+const App = () => {
+    const [visible, setVisible] = React.useState(false);
+    const [eventTitle, setEventTitle] = React.useState('');
+    const [description, setDescription] = React.useState('');
+    const [isDatePickerVisible, setDatePickerVisibility] = React.useState(false);
+
+    const handleSave = () => {
+        console.log('Event Title:', eventTitle, description); //just to make sure event title is saved
+        setVisible(false);
+    }
+    const handleCancel = () => {
+        setEventTitle(''); //set it to empty if user entered something but cancelled form
+        setVisible(false);
+    }
+
+    return (
+        <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
```

Comment:
Remove this wrapper view. This whole component file is dedicated to the modal only. So it's basically not a separate `App`.